sharding-core/sharding-core-route/src/main/java/org/apache/shardingsphere/core/route/router/sharding/ParsingSQLRoutingResultChecker.java

```diff
@@ -0,0 +1,212 @@
+        if (shardingStatement instanceof ShardingDropIndexOptimizedStatement) {
+            return;
+        }
+        Multimap<RoutingUnit, TableUnit> absentRoutingUnitMap = getAbsentRoutingUnit(routingResult.getRoutingUnits());
+        if (absentRoutingUnitMap.isEmpty()) {
+            return;
+        }
+        if (routingEngine instanceof StandardRoutingEngine) {
+            throwAbsentStandardRoutingResultException(shardingStatement, absentRoutingUnitMap);
+        } else {
+            throwOthersRoutingResultException(absentRoutingUnitMap);
+        }
+    }
+    
+    /**
+     * Throw other's absent routing result exception.
+     *
+     * @param absentRoutingUnitMap absent routingUnit map
+     */
+    private void throwOthersRoutingResultException(final Multimap<RoutingUnit, TableUnit> absentRoutingUnitMap) {
+        RoutingUnit routingUnit = absentRoutingUnitMap.keySet().iterator().next();
+        Collection<String> absentDataNodes = Lists.newArrayListWithExpectedSize(absentRoutingUnitMap.get(routingUnit).size());
+        for (TableUnit each : absentRoutingUnitMap.get(routingUnit)) {
+            absentDataNodes.add(routingUnit.getDataSourceName() + "." + each.getActualTableName());
+        }
+        if (!absentDataNodes.isEmpty()) {
+            throwExceptionForAbsentDataNode(absentDataNodes, "");
+        }
+    }
+    
+    /**
+     * Throw StandardRoutingEngine's absent routing result exception.
+     *
+     * @param shardingStatement    sharding statement
+     * @param absentRoutingUnitMap absent routing unit map
+     */
+    private void throwAbsentStandardRoutingResultException(final ShardingOptimizedStatement shardingStatement, final Multimap<RoutingUnit, TableUnit> absentRoutingUnitMap) {
+        RoutingUnit routingUnit = absentRoutingUnitMap.keySet().iterator().next();
+        Collection<String> absentDataNodes = Lists.newArrayListWithExpectedSize(absentRoutingUnitMap.get(routingUnit).size());
+        ShardingStrategyConfiguration databaseStrategy = null;
+        ShardingStrategyConfiguration tableStrategy = null;
+        if (null != shardingRule.getRuleConfiguration().getDefaultDatabaseShardingStrategyConfig()) {
+            databaseStrategy = shardingRule.getRuleConfiguration().getDefaultDatabaseShardingStrategyConfig();
+        }
+        if (null != shardingRule.getRuleConfiguration().getDefaultTableShardingStrategyConfig()) {
+            tableStrategy = shardingRule.getRuleConfiguration().getDefaultTableShardingStrategyConfig();
+        }
+        for (TableUnit each : absentRoutingUnitMap.get(routingUnit)) {
+            absentDataNodes.add(routingUnit.getDataSourceName() + "." + each.getActualTableName());
+            Optional<TableRuleConfiguration> tableRuleConfiguration = getTableRuleConfiguration(shardingRule.getRuleConfiguration().getTableRuleConfigs(), each.getLogicTableName());
+            if (!tableRuleConfiguration.isPresent()) {
```

Comment:
Better this way:

            if (tableRuleConfiguration.isPresent()) {
                databaseStrategy = tableRuleConfiguration.get().getDatabaseShardingStrategyConfig();
                tableStrategy = tableRuleConfiguration.get().getTableShardingStrategyConfig();
            }